### ISSUE TITLE
Effective en-passant

### ISSUE THREAD

**Issue by yo35** [yo35/kokopu#31]
Currently, the en-passant flag of a `Position` (see [`Position.enPassant()`](https://kokopu.yo35.org/docs/3.0.0/classes/Position.html#enPassant)) corresponds to the column on which a 2-square pawn move has just happen, if any.

Among other things, this flag is used:
1) to decide whether a position is legal or not (e.g. if the flag is set to file `d` and Black is to play, then there must be a white pawn in d4 and two empty squares in d3 and d2 for the position to be considered as legal, see [`Position.isLegal()`](https://kokopu.yo35.org/docs/3.0.0/classes/Position.html#isLegal)),
2) to generate the FEN corresponding a chess position (see [`Position.fen()`](https://kokopu.yo35.org/docs/3.0.0/classes/Position.html#fen)),
3) to compare two positions (see [`Position.isEqual()`](https://kokopu.yo35.org/docs/3.0.0/classes/Position.html#isEqual)).

As discussed in #27 (at the end), this leads to unexpected behaviors in some cases:

```
const p = new Position();
p.play('e4');
console.log(p.fen())
// rnbqkbnr/pppppppp/8/8/4P3/8/PPPP1PPP/RNBQKBNR b KQkq e3 0 1 (current behavior)
// rnbqkbnr/pppppppp/8/8/4P3/8/PPPP1PPP/RNBQKBNR b KQkq - 0 1 (what would be expected,
// and also what is obtained with lichess, scid...)
```

```
const p1 = new Position();
p1.play('e4');
p1.play('e5');
p1.play('d4');
const p2 = new Position();
p2.play('d4');
p2.play('e5');
p2.play('e4');
console.log(Position.isEqual(p1, p2));
// -> false (current behavior), although the positions are the same in terms of
// board state and possible moves.
```

Moreover, the fact that the en-passant flag is used to decide whether a position is legal or not may cause some issues when processing FEN strings issued by unreliable sources (e.g. a position could be judged as illegal by Kokopu just because of a en-passant flag incorrectly set in the input FEN).

**Comment by yo35** [comment]
To fix this, a so-called "effective en-passant" flag will be derived from the en-passant flag and maintained internally within each `Position` instance:

- "effective en-passant" will correspond to the column on which en-passant capture is actually possible, if any.
- "effective en-passant" will be a computed attribute (no way to set it directly).
- "effective en-passant" will be used for FEN generation and `Position` instance comparison.
- the en-passant flag will no longer affect the legal status of a `Position`: instead, if the en-passant flag is set to a column on which it is not possible that a 2-square pawn has just happen, the effective en-passant flag will be set to `'-'` (i.e. "no en-passant capture is allowed").

**Comment by yo35** [comment]
@mbutsykin FYI

**Comment by yo35** [comment]
Available in 3.1.0.
